### PR TITLE
add support for IAM roles

### PR DESCRIPTION
**Pull request by SaurabhBajaj**
I would love to add support for IAM roles here since that's a more secure way of managing AWS credentials, without generating aws keys for the autoscaler. Boto3 session accepts None values for the 
credentials (http://boto3.readthedocs.io/en/latest/reference/core/session.html) in which case it would traverse it's credentials fetching logic, as mentioned in the following link: http://boto3.readthedocs.io/en/latest/guide/configuration.html
@cberner

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -58,30 +59,35 @@
...
                       aws_secret_key=aws_secret_key,
+                      use_aws_iam_role=use_aws_iam_role,
                       aws_regions=aws_regions_list,
                       azure_client_id=azure_client_id,
                       azure_client_secret=azure_client_secret,
                       azure_subscription_id=azure_subscription_id,
                       azure_tenant_id=azure_tenant_id,
-                      azure_resource_group_names=azure_resource_groups.split(',') if azure_resource_groups else [],
-                      azure_slow_scale_classes=azure_slow_scale_classes.split(',') if azure_slow_scale_classes else [],
+                      azure_resource_group_names=azure_resource_groups.split(
```
Comment (cberner) [inline main.py]: Looks like this is just white-space changes. Could you revert them, if so?

File: main.py [hunk 2]
```diff
@@ -58,30 +59,35 @@
...
-    logger_handler.setFormatter(logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s', datefmt='%Y-%m-%dT%H:%M:%S%z'))
+    logger_handler.setFormatter(logging.Formatter(
+        '%(asctime)s - %(name)s - %(levelname)s - %(message)s', datefmt='%Y-%m-%dT%H:%M:%S%z'))
     logger.addHandler(logger_handler)
     logger.setLevel(DEBUG_LOGGING_MAP.get(verbose, logging.CRITICAL))
 
     aws_regions_list = aws_regions.split(',') if aws_regions else []
     if not (aws_secret_key and aws_access_key) and aws_regions_list:
-        logger.error("Missing AWS credentials. Please provide aws-access-key and aws-secret-key.")
+        logger.error(
```
Comment (cberner) [inline main.py]: Looks like this is just white-space changes. Could you revert them, if so?

File: main.py [hunk 3]
```diff
@@ -58,30 +59,35 @@
...
 def main(cluster_name, aws_regions, azure_resource_groups, azure_slow_scale_classes, sleep, kubeconfig,
          azure_client_id, azure_client_secret, azure_subscription_id, azure_tenant_id,
-         aws_access_key, aws_secret_key, pod_namespace, datadog_api_key,
+         aws_access_key, aws_secret_key, use_aws_iam_role, pod_namespace, datadog_api_key,
          idle_threshold, type_idle_threshold, max_scale_in_fraction, drain_utilization,
          over_provision, instance_init_time, no_scale, no_maintenance,
          slack_hook, slack_bot_token, dry_run, verbose):
     logger_handler = logging.StreamHandler(sys.stderr)
-    logger_handler.setFormatter(logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s', datefmt='%Y-%m-%dT%H:%M:%S%z'))
+    logger_handler.setFormatter(logging.Formatter(
```
Comment (cberner) [inline main.py]: Looks like this is just white-space changes. Could you revert them, if so?

**Comment by SaurabhBajaj** [comment]
@cberner  updated the code to remove the autoformatted white space changes. Do you know what might be causing the azure_api.py error? 

```
The command "nosetests test/" exited with 0.
2.48s$ mypy --ignore-missing-imports autoscaler/azure_api.py
autoscaler/azure_api.py:229: error: Item "None" of "Optional[AzureScaleSet]" has no attribute "capacity"
The command "mypy --ignore-missing-imports autoscaler/azure_api.py" exited with 1.
```

**Comment by SaurabhBajaj** [comment]
@cberner  how would the docker images in quay.io be updated after a branch in merged?

**Comment by cberner** [comment]
Resolved the typing test failure and merged. Thanks!

**Comment by cberner** [comment]
Releasing 1.4.0, which will include this change

**Comment by SaurabhBajaj** [comment]
Thanks @cberner ! 🙏

**Pull request closed without merging**